src/nvim/api/buffer.c

```diff
@@ -1683,7 +1683,7 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id,
   }
 
   if (col2 >= 0) {
-    if (line2 >= 0 && line2 < buf->b_ml.ml_line_count) {
+    if (!ephemeral && line2 >= 0 && line2 < buf->b_ml.ml_line_count) {
```

Comment:
IMHO a separate `if (ephemeral) { len = MAXCOL; } ` branch is cleaner

Reply:
maybe could repeat the same pattern here? :)

Reply:
Yup